soccer/SystemState.hpp

```diff
@@ -67,7 +68,7 @@ class Context;
  */
 class SystemState {
 public:
-    SystemState(Context* const context);
+    SystemState(Context* context);
```

Comment:
This to me removes context about the variable `context` and how it's to be used. If it's complaining that  `Robot`'s constructor isn't treating const correctly, I would rather fix that than remove context here.

Reply:
If it's more than one or two lines, it's not a big deal and can be fixed another time

Reply:
This is a "ptr to `const Context`" vs "`const` ptr to `Context`" issue I think. It's currently a `const` ptr to `Context`", so the `const` is superfluous. Should this be a "ptr to `const Context`?

Reply:
It's supposed to be const ptr to Context. If it's superfluous, nevermind then. I guess I'm still stuck on the const everything that shouldn't change train lol

Reply:
yeah... it's superfluous the same way as like how a `const float` parameter is superfluous....